core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -812,6 +812,26 @@ object ReassignPartitionsCommand extends Logging {
     proposedAssignments
   }
 
+  private def describeTopics(adminClient: Admin,
+                             topics: Set[String])
+                             : Map[String, TopicDescription] = {
+    adminClient.describeTopics(topics.asJava).values.asScala.map {
+      case (topicName, topicDescriptionFuture) =>
+        try {
+          topicName -> topicDescriptionFuture.get
+        }
+        catch {
+          case t: ExecutionException =>
+            if (classOf[UnknownTopicOrPartitionException].isInstance(t.getCause)) {
```

Comment:
Hmm. why not write this as `t.getCause.isInstanceOf[UnknownTopicOrPartitionException]`?

Reply:
Also, you can have the `if` in the `case t` line and then a second `case` for the rethrow case.

Reply:
I initially used `t.getCause != null && t.getCause.isInstanceOf[UnknownTopicOrPartitionException]`. @chia7712 suggested to use `classOf[UnknownTopicOrPartitionException].isInstance(t.getCause)` to avoid having to do the null check as `isInstance` does it. That seemed reasonable to me so I went with it. Is there a reason not to use it?

I do agree with your second comment.

Reply:
Hmm, I think this is an example of code that is less readable. If `cause` may be nullable, it's better to write code that makes that clear rather than a non obvious alternative that could have been used for many other reasons (using `Option` to handle nullables is fine as a counter example since it's common usage to do that).

What do you think?

Reply:
Oh, one more thing, did you check that the null check is actually needed? I think isInstanceOf is defined for null too.

Reply:
> I think isInstanceOf is defined for null too.

you are right. Scala does define it. https://scala-lang.org/files/archive/spec/2.13/spec.pdf

‘’’

isInstanceOf[T] always returns false.
‘’’

Good to know that :)

+1 to use ‘t.getCause.isInstanceOf[UnknownTopicOrPartitionException]‘ as it can deal with null check.

Sorry for my imprecise comment :(

Reply:
@ijuma Yeah, I do agree with you. For the null check, I was not aware that isInstanceOf handles it. That's good to know, thanks.

I will open a small PR to fix this.